### ISSUE TITLE
RefreshAllMaterializedViews et ordre de rafraichissement des VM

### ISSUE THREAD

**Issue by camillemonchicourt** [PnEcrins/GeoNature-atlas#97]
La fonction `refreshallmaterializedviews` de la BDD permet de lister toutes les vues matérialisées d'un schéma et de les rafraîchir. 
Cependant nous n'avons pas la main sur l'ordre dans lequel elles sont rafraîchit.

`SELECT matviewname FROM pg_matviews` les retourne dans cet ordre :

- vm_mois
- vm_taxref
- vm_observations
- vm_medias
- vm_taxons_plus_observes
- vm_communes
- vm_altitudes
- vm_observations_mailles
- vm_cor_taxon_attribut
- vm_search_taxon
- vm_taxons

Il est important que `vm_observations` soit la première rafraîchie car les autres en dépendent. 
Or cela ne semble pas être le cas. 

Premier rafraîchissement : 

![1ere heure](https://cloud.githubusercontent.com/assets/4418840/21975256/e7d65142-dbcb-11e6-9223-a0c1d37d7966.jpg)

Deuxième rafraîchissement : 

![2eme heure](https://cloud.githubusercontent.com/assets/4418840/21975262/f08d7040-dbcb-11e6-9e62-537346190f4e.jpg)

On voit ici que `vm_observations` est rafraîchie après `vm_taxons` (nb d'observations) mais aussi après `vm_altitudes`. Par contre `vm_mois` est bien rafraîchie après `vm_observations`.

Du coup il faut lancer 2 fois de suite la fonction pour que les données soient vraiment rafraîchies et cohérentes.

**Comment by gildeluermoz** [comment]
Il faut réanalyser l’enchaînement des fonctions. En effet, si elles sont faites dans cet ordre, ce n'est pas par hasard. Chaque vue est dépendante des autres.
``_Il est important que vm_observations soit la première rafraîchie car les autres en dépendent.
Or cela ne semble pas être le cas._``
Par exemple si on ajoute un taxon, il faut bien rafraîchir ``vm_taxons`` d'abord si on veut ensuite faire remonter les observations de ce taxon dans ``vm_observations``.
Donc si des difficultés subsistent, il faut refaire une analyse plus fine.
A creuser.

**Comment by camillemonchicourt** [comment]
Non, on ne liste pas les VM dans la fonction, elle le fait elle-même donc on n'a pas la main sur l'ordre de rafraîchissement des vues. 
C'est bien car si on ajoute une VM la fonction la prendra automatiquement en compte. Mais du coup ça ne permet pas de définir un ordre. 

Et non plus, `vm_observations` ne se base pas sur `vm_taxons` mais l'inverse. La liste des taxons se construit en fonction des taxons existants dans la VM des observations.
C'est bien `vm_observations` qui est à rafraichir en tout premier
